### PR TITLE
Allow creation of images in clouds without s3 support

### PR DESCRIPTION
**Pull request by schirevko**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ description: |-
 # VKCS Provider's changelog
 
 #### v0.1.13 (unreleased)
+- Added retry of creating resource_vkcs_images_image with default store if initial creation failed
```
Comment (ftersin) [inline CHANGELOG.md:11]: Allow create images in clouds w/o s3 support.

File: vkcs/resource_vkcs_images_image.go [hunk 2]
```diff
@@ -219,6 +221,62 @@ func resourceImagesImage() *schema.Resource {
...
+	// upload
+	imgFile, err = os.Open(imgFilePath)
+	if err != nil {
+		return newImg.ID, fileChecksum, &imagePrepareUploadError{fmt.Sprintf("Error opening file %q: %s", imgFilePath, err)}
+	}
+	defer imgFile.Close()
+	log.Printf("[WARN] Uploading image %s (%d bytes). This can be pretty long.", d.Id(), fileSize)
+
+	res := imagedata.Upload(imageClient, newImg.ID, imgFile)
+	if res.Err != nil {
```
Comment (ftersin) [inline vkcs/resource_vkcs_images_image.go:273]: could we be more selective with the error to skip cases like 4xx, 503 etc

File: CHANGELOG.md (hunk 1 above)
Comment (schirevko) [reply CHANGELOG.md]: Rewritten

File: vkcs/resource_vkcs_images_image.go (hunk 2 above)
Comment (schirevko) [reply vkcs/resource_vkcs_images_image.go]: Expanded error handling to retry creation only if we got 500 error

File: vkcs/images_image.go [hunk 3]
```diff
@@ -165,11 +167,19 @@ func resourceImagesImageExpandProperties(v map[string]interface{}) map[string]st
 			properties[key] = v
 		}
 	}
-	properties["store"] = "s3"
 
 	return properties
 }
 
+func resourceImagesImageNeedsDefaultStore(endpoint string) bool {
```
Comment (gmmephisto) [inline vkcs/images_image.go:174]: We should check that domain contains mask only in the host part of endpoint URL

File: vkcs/images_image.go [hunk 4]
```diff
@@ -19,6 +19,8 @@ import (
 	"github.com/gophercloud/gophercloud/openstack/imageservice/v2/images"
 )
 
+var imagesDefaultStoreEndpointMasks = []string{"devmail.ru", "ams."}
```
Comment (gmmephisto) [inline vkcs/images_image.go:22]: Maybe rewrite as RE?
* `*.devmail.ru$`
* `^ams.*`

File: vkcs/resource_vkcs_images_image.go [hunk 5]
```diff
@@ -231,6 +231,9 @@ func resourceImagesImageCreate(ctx context.Context, d *schema.ResourceData, meta
 
 	properties := d.Get("properties").(map[string]interface{})
 	imageProperties := resourceImagesImageExpandProperties(properties)
+	if !resourceImagesImageNeedsDefaultStore(imageClient.Endpoint) {
+		imageProperties["store"] = "s3"
```
Comment (gmmephisto) [inline vkcs/resource_vkcs_images_image.go:235]: Maybe move `"s3"` to well-named constants?

**Review by gmmephisto — CHANGES_REQUESTED** [review]

File: vkcs/images_image.go (hunk 3 above)
Comment (schirevko) [reply vkcs/images_image.go:176]: Rewrote to check only the host part

File: vkcs/images_image.go (hunk 4 above)
Comment (schirevko) [reply vkcs/images_image.go]: Rewrote

File: vkcs/resource_vkcs_images_image.go (hunk 5 above)
Comment (schirevko) [reply vkcs/resource_vkcs_images_image.go]: Moved

**Review by gmmephisto — APPROVED** [review]

**Pull request merged**